test/integration/docker_test.go

```diff
@@ -158,3 +163,81 @@ func TestForceSystemdEnv(t *testing.T) {
+	startResult, err := Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to start minikube with args: %q : %v", startResult.Command(), err)
+	}
+	time.Sleep(time.Second * 10)
+
+	// if we are in a shell, we need to run 'ssh-agent bash' to enable the ssh-agent
+	// but if we are in an integration test, we don't have a bash, so we need to get the environment variables set by ssh-agent, and use them in the following tests
+	result, err := Run(t, exec.CommandContext(ctx, "ssh-agent"))
+	if err != nil {
+		t.Errorf("failed to execute ssh-agent bash, error: %v, output: %s", err, result.Output())
+	}
+	output := result.Output()
+	// get SSH_AUTH_SOCK
+	groups := regexp.MustCompile(`SSH_AUTH_SOCK=(\S*);`).FindStringSubmatch(output)
+	if len(groups) < 2 {
+		t.Errorf("failed to acquire SSH_AUTH_SOCK, output is %s", output)
+	}
+	sshAuthSock := groups[1]
+	// get SSH_AGENT_PID
+	groups = regexp.MustCompile(`SSH_AGENT_PID=(\S*);`).FindStringSubmatch(output)
+	if len(groups) < 2 {
+		t.Errorf("failed to acquire SSH_AUTH_SOCK, output is %s", output)
+	}
+	sshAgentPid := groups[1]
+
+	// execute 'minikube ssh-host --append-known'
+	result, err = Run(t, exec.CommandContext(ctx, Target(), "ssh-host", "--append-known", "-p", profile))
+	if err != nil {
+		t.Errorf("failed to execute 'minikube ssh-host --append-known', error: %v, output: %s", err, result.Output())
+	}
+
+	// execute 'minikube docker-env --ssh-host --ssh-add' and extract the 'DOCKER_HOST' environment value
+	cmd = exec.CommandContext(ctx, "/bin/bash", "-c", fmt.Sprintf("SSH_AUTH_SOCK=%s SSH_AGENT_PID=%s %s docker-env --ssh-host --ssh-add -p %s", sshAuthSock, sshAgentPid, Target(), profile))
+	result, err = Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to execute minikube docker-env --ssh-host --ssh-add, error: %v, output: %s", err, result.Output())
+	}
+
+	output = result.Output()
+	groups = regexp.MustCompile(`DOCKER_HOST="(\S*)"`).FindStringSubmatch(output)
+	if len(groups) < 2 {
+		t.Errorf("failed to acquire SSH_AUTH_SOCK, output is %s", output)
+	}
+	dockerHost := groups[1]
+
+	cmd = exec.CommandContext(ctx, "/bin/bash", "-c", fmt.Sprintf("SSH_AUTH_SOCK=%s SSH_AGENT_PID=%s DOCKER_HOST=%s docker version", sshAuthSock, sshAgentPid, dockerHost))
+
+	result, err = Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to execute 'docker version',error: %v, output:%s", err, result.Output())
+	}
+	// if we are really connecting to nerdctld inside node, docker version output should have word 'nerdctl'
+	if !strings.Contains(result.Output(), "nerdctl") {
+		t.Error("failed to detect keyword 'nerdctl' in output of docker version")
+	}
+
+	// now try to build an image
+	cmd = exec.CommandContext(ctx, "/bin/bash", "-c", fmt.Sprintf("SSH_AUTH_SOCK=%s SSH_AGENT_PID=%s DOCKER_HOST=%s docker build -t test:latest testdata/docker-env", sshAuthSock, sshAgentPid, dockerHost))
+	result, err = Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to build images, error: %v, output:%s", err, result.Output())
+	}
```

Comment:
"minikube image ls --format=json"

or 
on linux and mac
$(minikube docker-env); docker image ls

and on windows ...
something else